synapse/handlers/message.py

```diff
@@ -1274,23 +1274,6 @@ async def handle_new_client_event(
                 )
                 return prev_event
 
-        if event.is_state() and (event.type, event.state_key) == (
-            EventTypes.Create,
-            "",
-        ):
-            room_version_id = event.content.get(
-                "room_version", RoomVersions.V1.identifier
-            )
-            maybe_room_version_obj = KNOWN_ROOM_VERSIONS.get(room_version_id)
-            if not maybe_room_version_obj:
-                raise UnsupportedRoomVersionError(
-                    "Attempt to create a room with unsupported room version %s"
-                    % (room_version_id,)
-                )
```

Comment:
I'm guessing something else earlier will reject the event when we attempt to create it? (So this code essentially is dead code.)

Reply:
sorry, I meant to say that. Yes, indeed.